src/qibolab/platforms/abstract.py

```diff
@@ -99,6 +99,13 @@ class Qubit:
     mixer_readout_g: float = 0.0
     mixer_readout_phi: float = 0.0
 
+    @property
+    def channels(self):
+        for name in ["readout", "feedback", "drive", "flux", "twpa"]:
+            channel = getattr(self, name)
+            if channel is not None:
+                yield channel
```

Comment:
Since `name` is only used to retrieve the attribute, and the attribute is just whatever `Channel` among the attributes, you can just iterate over fields:
```suggestion
        for field in dataclasses.fields(self):
            if isinstance(field, Channel):
                yield field
```

In this way you do not depend on attribute names, and it should be simpler to extend (but has to be tested).